lib/node/process.dart

```diff
@@ -180,3 +176,57 @@ class ProcessResult {
 
   String toString() => '${exit}';
 }
+
+/// Shell out to query the given environment variable. This is slower but more
+/// reliable than the node env map.
+String queryEnv(String variable) {
+  try {
+    return execSync('echo \$$variable');
+  } catch (e) {
+    return null;
+  }
+}
+
+/// This class exists to help manage situations where Atom is running in an
+/// environment without a properly set up environment (env and PATH variables).
+class MacShellWrangler {
```

Comment:
A little more here would be helpful. /bin/sh does not work but /bin/bash does? Problem exists only on Macs? Just need a clear description of the problem encountered and the solution implemented.